### PR TITLE
Add check for pre-specified YACC to yacc tool's exist()

### PR DESCRIPTION
**Pull request by bdbaddog**
Fix exist()

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `CHANGES.txt` (and read the `README.rst`)
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by acmorrow** [comment]
LGTM

File: SCons/Tool/yacc.py [hunk 1]
```diff
@@ -162,6 +162,9 @@ def generate(env):
 
 
 def exists(env):
+    if 'YACC' in env:
```
Comment (acmorrow) [inline SCons/Tool/yacc.py:165]: Is it a good idea to include a check that `YACC` is something truthy as well?

Comment (acmorrow) [reply SCons/Tool/yacc.py:165]: Or will Detect just deal with that for you?

Comment (bdbaddog) [reply SCons/Tool/yacc.py:165]: If YACC is garbage Detect will be False (or throw an exception if it's not understandable garbage I suppose)

Comment (acmorrow) [reply SCons/Tool/yacc.py:165]: Sounds good. LGTM for sure then. Thanks for the quick fix.

**Comment by bdbaddog** [comment]
Merging.

**Pull request merged**